### PR TITLE
4614 number inputs

### LINKED ISSUE #4614: Add The Support Of The Custom Type For Input Field
**Description**:
At the moment we have the [Field](https://github.com/scandipwa/scandipwa/tree/master/packages/scandipwa/src/component/Field) component that is configurable and gives to us the ability to render any field.
In order to pass different fields, we can use the `type` prop where we pass any of the `FIELD_TYPE` values, and depending on the [renderer](https://github.com/scandipwa/scandipwa/blob/master/packages/scandipwa/src/component/Field/Field.component.js#L83-L105), this component will render the appropriate field.
But unfortunately, the current [implementation](https://github.com/scandipwa/scandipwa/blob/master/packages/scandipwa/src/component/Field/Field.component.js#L330) doesn't give us the ability to render field types that don't have any rendered. Thus some default HTML inputs aren't supported by ScandiPWA.

**Implementation**:
1. Move current `[type="number"]` styles for `NumberField` to `NumberField` styles and fix the selector, make it more specific.
2. In the `Field` component add a check. If rendered is undefined, then call the `renderDefaultInput`.
3. Rename the `FIELD_TYPE.number` to `FIELD_TYPE.numberWithControls` and change it all over the project.
4. Add the `number` option to the `FIELD_TYPE` in order to have the ability to render the regular number inputs.
5. Add `tel` option to `FIELD_TYPE` in order to have the ability to render the phone number inputs.
6. Change phone number fields inside the address form (order and my account pages), set the type `tel`.

**Test requirements**
Since we don't have a real number field on the project and QAs cannot really test it, after you finish, please contact QA and provide a demo, where you add the regular number field to the form and it works.

### LINKED ISSUE #4635: Ooops page when opening /styleguide page
**Description**:
Ooops page when opening /styleguide page

**Expected behavior**:
/styleguide page opens

**To Reproduce**:
1. Open Style guide https://scandipwapmrev.indvp.com/default/styleguide

**Actual result**:
Ooops page
Error message 'TypeError: Cannot read properties of undefined (reading 'isAvailable')' appears in the console.
Screenshot 2022-03-31 at 11 11 29

### PR DESCRIPTION
**Pull request by AzizKHAN030**
**Related issue(s):**
* Fixes #4614 
* Fixes #4635

**Problem:**
* Regular inputs with types 'number' and 'tel' were not supported

**In this PR:**
* Moved styles for custom FieldNumber with controls, added support for regular number and tel types, added check if FIELD_TYPE is undefined then should be rendered default field

Demo video for **QA** : [https://drive.google.com/file/d/1YzUZwm7omewxLxkGnuP99kBM5HDE3cxG/view?usp=sharing](url)

### CODE REVIEW

**Comment by AzizKHAN030** [comment]
For **QA**: Needs proper testing of quantity fields with controls

**Comment by aleksandrakorolova** [comment]
Hey
Please, add input fields with input type ```number``` and ```tel``` on our /styleguide page but before it fix it. Now ooops page opens 
Error message 'TypeError: Cannot read properties of undefined (reading 'isAvailable')' appears in the console.
Screenshot 2022-03-31 at 11 11 29

**Comment by AzizKHAN030** [comment]
Seems like Amplify stuck, but the changes are applied

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
1. /styleguide page is **fixed**
2. Please, add all views for tel and number fields (enable and disabled fields, required and not required, validated and etc) as text/dropdown/checkbox/radio buttons
Screenshot 2022-04-12 at 09 03 35

File: packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js [hunk 1]
```diff
@@ -366,6 +366,139 @@ export class StyleGuidePageComponent extends PureComponent {
...
+                      <Field
+                        type={ FIELD_TYPE.number }
+                        label="Number"
+                        attr={ {
+                            placeholder: __('Number'),
+                            step: 5,
+                            min: 0
+                        } }
+
+                      />
```
Comment (zans-laksa) [inline packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js:399]: Please keep the same indents for the adjacent elements in the markup.

File: packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js [hunk 2]
```diff
@@ -366,6 +366,139 @@ export class StyleGuidePageComponent extends PureComponent {
...
+                  attr={ {
+                      placeholder: __('Number'),
+                      block: 'StyleGuidePage',
+                      elem: 'Text',
+                      mods: { isValid: true },
+                      value: 56,
+                      step: 5,
+                      min: 0
+                  } }
+
```
Comment (zans-laksa) [inline packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js:441]: Please remove the empty line.

File: packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js [hunk 3]
```diff
@@ -366,6 +366,139 @@ export class StyleGuidePageComponent extends PureComponent {
...
+                      } }
+                      isDisabled
+                    />
+                      <Field
+                        type={ FIELD_TYPE.tel }
+                        label="Telephone"
+                        attr={ {
+                            placeholder: __('Telephone')
+                        } }
+                      />
```
Comment (zans-laksa) [inline packages/scandipwa/src/route/StyleGuidePage/StyleGuidePage.component.js:466]: Please fix indents here too.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

**Review by zans-laksa — APPROVED** [review]

**Review by zans-laksa — APPROVED** [review]

**Review by zans-laksa — APPROVED** [review]

**Pull request merged**